cloudfoundry/managers/session.go

```diff
@@ -221,30 +225,35 @@ func (s *Session) init(config *configv3.Config, configUaa *configv3.Config, conf
 	var accessToken string
 	var refreshToken string
 	var errType string
+	var goClientConfigOptions goConfig.Option
 
 	tokFromStore := s.loadTokFromStoreIfNeed(configSess.StoreTokensPath, uaaClient.RefreshAccessToken)
 	if tokFromStore.IsSet() {
 		accessToken = tokFromStore.AccessToken
 		refreshToken = tokFromStore.RefreshToken
+		goClientConfigOptions = goConfig.Token(accessToken, refreshToken)
 	} else if configSess.SSOPasscode != "" {
 		// try connecting with SSO passcode to retrieve access token and refresh token
 		accessToken, refreshToken, err = uaaClient.Authenticate(map[string]string{
 			"passcode": configSess.SSOPasscode,
 		}, configSess.Origin, constant.GrantTypePassword)
 		errType = "SSO passcode"
+		goClientConfigOptions = goConfig.Token(accessToken, refreshToken)
 	} else if config.CFUsername() != "" {
 		// try connecting with pair given on uaa to retrieve access token and refresh token
 		accessToken, refreshToken, err = uaaClient.Authenticate(map[string]string{
 			"username": config.CFUsername(),
 			"password": config.CFPassword(),
 		}, configSess.Origin, constant.GrantTypePassword)
 		errType = "username/password"
+		goClientConfigOptions = goConfig.UserPassword(config.CFUsername(), config.CFPassword())
```

Comment:
Why not use the token like above ?

Maybe it's best to get the token with the old api for now, and set the goClientConfigOptions at the end with the token like you do line 241.

Reply:
indeed.